WordPress/gutenberg
# [TypeScript] Simplify core-data types

## What?

This PR prepares core-data selector types for the [newly supported useSelect's autocompletion](https://github.com/WordPress/gutenberg/pull/41911).

Technically, it renames selectors.ts to selectors.js and removes the ["TypeScript lookup table" magic](https://github.com/WordPress/gutenberg/pull/41235).

## Why?

The current types heavily rely on generics, and [generics don't play well with useSelect](https://github.com/WordPress/gutenberg/pull/41578) which is a blocker for bringing autocompletion support for core-data selectors.

Therefore, I'd like to take a step back and start again, smaller. Here's a hypothetical progression to illustrate what I mean:

1. Autocompletion for selectors
2. Autocompletion for actions
3. Autocompletion inside registry selectors
4. Autocompletion for thunks
5. Stricter types for some selectors


## Testing Instructions

1. Confirm the checks are green (hint: they're not yet, docgen strikes again)
7. Read the change and confirm the types specified provided in JSDoc comments are sound.

cc @dmsnell @sarayourfriend
